src/services/StorageService/StorageService.ts

```diff
@@ -1,124 +1,82 @@
+    }
+    return this.instance;
+  }
 
-    const region = ((await AsyncStorage.getItem(Key.Region)) as Region | undefined) || undefined;
-    this.region.set(region);
+  private unsecureKeyValueStore: KeyValueStore;
+  private secureKeyValueStore: KeyValueStore;
 
-    const onboardedDatetimeStr =
-      ((await AsyncStorage.getItem(Key.OnboardedDatetime)) as string | undefined) || undefined;
-    const onboardedDatetime = onboardedDatetimeStr ? new Date(onboardedDatetimeStr) : undefined;
-    this.onboardedDatetime.set(onboardedDatetime);
+  private serialPromiseQueue: PQueue;
 
-    const forceScreen = ((await AsyncStorage.getItem(Key.ForceScreen)) as ForceScreen | undefined) || undefined;
-    this.forceScreen.set(forceScreen);
+  constructor(unsecureKeyValueStore: KeyValueStore, secureKeyValueStore: KeyValueStore) {
+    this.unsecureKeyValueStore = unsecureKeyValueStore;
+    this.secureKeyValueStore = secureKeyValueStore;
+    this.serialPromiseQueue = new PQueue({concurrency: 1});
+  }
 
-    const skipAllSet = (await AsyncStorage.getItem(Key.SkipAllSet)) === '1';
-    this.skipAllSet.set(skipAllSet);
+  save(keyDefinition: KeyDefinition, value: string): Promise<void> {
+    return this.serialPromiseQueue.add(() => {
+      switch (keyDefinition.storageType) {
+        case StorageType.Unsecure:
+          return this.unsecureKeyValueStore.save(keyDefinition.keyIdentifier, value);
+        case StorageType.Secure:
+          return this.secureKeyValueStore.save(keyDefinition.keyIdentifier, value);
+      }
+    });
+  }
 
-    const userStopped = (await AsyncStorage.getItem(Key.UserStopped)) === '1';
-    this.userStopped.set(userStopped);
+  retrieve(keyDefinition: KeyDefinition): Promise<string | null> {
+    return this.serialPromiseQueue.add(() => {
+      switch (keyDefinition.storageType) {
+        case StorageType.Unsecure:
+          return this.unsecureKeyValueStore.retrieve(keyDefinition.keyIdentifier);
+        case StorageType.Secure:
+          return this.secureKeyValueStore.retrieve(keyDefinition.keyIdentifier);
+      }
+    });
+  }
 
-    const hasViewedQrInstructions = (await AsyncStorage.getItem(Key.HasViewedQrInstructions)) === '1';
-    this.hasViewedQrInstructions.set(hasViewedQrInstructions);
+  delete(keyDefinition: KeyDefinition): Promise<void> {
+    return this.serialPromiseQueue.add(() => {
+      switch (keyDefinition.storageType) {
+        case StorageType.Unsecure:
+          return this.unsecureKeyValueStore.delete(keyDefinition.keyIdentifier);
+        case StorageType.Secure:
+          return this.secureKeyValueStore.delete(keyDefinition.keyIdentifier);
+      }
+    });
+  }
 
-    const qrEnabled = (await AsyncStorage.getItem(Key.QrEnabled)) === '1';
-    this.qrEnabled.set(qrEnabled);
-  };
+  deteleAll(): Promise<void> {
+    return this.serialPromiseQueue.add(() => {
+      return this.unsecureKeyValueStore.deleteAll();
```

Comment:
Should this also include something like `this.secureKeyValueStore.deleteAll()`

Reply:
Unfortunately there is no existing API in RNSecureKeyStore to clear the storage. This feature is only accessible to QA through the debug menu and it looks like it is enough for them to reset the state of the app at this point. Definitely something we need to keep in mind if we start using the secure storage to keep `state` type of information.